### PR TITLE
EVG-5486 show build baron plugin when there are unprocessed change points

### PR DESCRIPTION
**Pull request by jimoleary**

### CODE REVIEW

File: public/static/app/perf-bb/ChangePointsService.js [hunk 1]
```diff
@@ -59,6 +59,18 @@ mciModule.factory('ChangePointsService', function(
...
 
+  // Get unprocessed change points for query, return promise
+  // :param query: The mongo query to filter on.
+  function getUnprocessedChangePoints(query){
+    return Stitch.use(STITCH_CONFIG.PERF).query(function(db) {
+      return db
+        .db(STITCH_CONFIG.PERF.DB_PERF)
+        .collection(STITCH_CONFIG.PERF.COLL_UNPROCESSED_POINTS)
+        .find(query)
+        .execute()
```
Comment (tychoish) [inline public/static/app/perf-bb/ChangePointsService.js]: oneline? semicolons?

Comment (jimoleary) [reply public/static/app/perf-bb/ChangePointsService.js]: In terms of semicolons, I would lean towards adding them but I was just following the conventions in this file (and it passed lint).

Personally I think multiline is probably a bit easier to read / edit.

**Comment by jimoleary** [comment]
@john-m-liu

I moved the code for change points into the ChangePointService  so that the stitch details are hidden from the panel (which makes sense).

I think that I'd prefer to only show the BB plugin when it is useful (so when there are failures and / or  unprocessed change points).

But there is more work to do. In evergreen, we need to create BFs for change points when there are no failures (the BF template needs to take this into account and the BF meta data needs to be set correctly) and we need to integrate change points into the task page better (ie. a change point failure status). I would like to do this in a separate ticket. 

This (small) change goes a long way towards making the creation of BFs for change points much more useful and easier.

**Comment by john-m-liu** [comment]
In that case could you make it a configuration option? Add it to the appropriate part of config.go. Don't worry about the UI to populate the setting - I can do that part

**Comment by bsamek** [comment]
Closing for the moment after talking to @jimoleary. He will re-open or open a new PR when ready.

**Pull request closed without merging**